test/system/src/test/java/io/pravega/test/system/ByteClientTest.java

```diff
@@ -0,0 +1,176 @@
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import mesosphere.marathon.client.MarathonException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class ByteClientTest extends AbstractSystemTest {
+
+    private static final String STREAM = "testByteClientStream";
+    private static final String SCOPE = "testByteClientScope" + RandomFactory.create().nextInt(Integer.MAX_VALUE);
+    private static final int PARALLELISM = 1;
+    private static final int MAX_PAYLOAD_SIZE = 100000000;
+    private static final int IO_ITERATIONS = 5;
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(8 * 60);
+
+    private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(4, "executor");
+    private final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(PARALLELISM);
+    private final StreamConfiguration config = StreamConfiguration.builder().scope(SCOPE)
+            .streamName(STREAM)
+            .scalingPolicy(scalingPolicy).build();
+    private URI controllerURI = null;
+    private StreamManager streamManager = null;
+    private ScheduledExecutorService writerExecutor = ExecutorServiceHelpers.newScheduledThreadPool(2, "byte-writer");
+    private ScheduledExecutorService readerExecutor = ExecutorServiceHelpers.newScheduledThreadPool(2, "byte-reader");
+
+
+    /**
+     * This is used to setup the services required by the system test framework.
+     *
+     * @throws MarathonException When error in setup.
+     */
+    @Environment
+    public static void initialize() throws MarathonException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+        streamManager = StreamManager.create(controllerURI);
+        assertTrue("Creating scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream", streamManager.createStream(SCOPE, STREAM, config));
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
+        ExecutorServiceHelpers.shutdown(writerExecutor);
```

Comment:
Closing the executor makes it so no other test can run. So this should be in an @ AfterClass function not an @ After function.

Reply:
I addressed this change. However, I realized that this pattern is followed by the majority of system tests: the shutdown of executors is in `@After` and they only declare one test method. If you believe that this patter is not correct, then we may need to change this in other system tests.